### DISCUSSION TITLE
Collections are hydrated with elements in the wrong order

### DISCUSSION THREAD

**Discussion by mi-dave** [livewire/livewire#7029] [7. Bugs (archived)]
### Livewire version

3.0.7

### Which PHP version are you using?

PHP 8.1

### Steps To Reproduce

```php
class Test extends Component
{
    public Collection $users;

    public function mount()
    {
        $this->users = User::orderBy('name')->get();
    }

    public function render()
    {
        return view('livewire.test');
    }

    public function doNothing()
    {
    }
}
```

```blade
<div>
    <ul>
        @foreach ($users as $user)
            <li>{{ $user->id }}. {{ $user->name }}</li>
        @endforeach
    </ul>

    <button wire:click="doNothing">Update</button>
</div>
```

```blade
<livewire:test />
```

Load the page - users are ordered by name.

Then click "Update" - users are now ordered by ID (or whatever order the database decided to return them).

### Additional information

I believe this was caused by the changes in #7003. In Livewire 3.0.5, they stayed in the correct order (but required multiple database queries).

### Please confirm (incomplete submissions will not be addressed)

- [X] I have provided easy and step-by-step instructions to reproduce the bug.
- [X] I have provided code samples as text and NOT images.
- [X] I understand my bug report will be removed if I haven't met the criteria above.

**Comment by MarcoCazzaro** [comment]
Check [this](https://github.com/livewire/livewire/pull/7003), I think you (we) must use Computed Properties now.

**Comment by joshhanley** [comment]
We've just fixed this issue in PR #7359. So should be available in the next release.
